app/controllers/LibAll.js

```diff
@@ -21,15 +21,20 @@ function renderAllLibrary(lib) {
       feedTemplate.renderFeedAsync(posts, options, callback);
     }
 
-    postModel.fetchPosts(
-      { repost: { $exists: false } },
-      null,
-      { after: options.after, before: options.before },
-      process,
-    );
+    const params = {
+      after: options.after,
+      before: options.before,
+    };
+    if (options.limit) params.limit = options.limit;
+
+    postModel.fetchPosts({ repost: { $exists: false } }, null, params, process);
   }
 
-  if (!feedTemplate.shouldRenderWholeProfilePage(options))
+  if (options.format == 'json')
+    renderFeed(function (feedHtml) {
+      lib.renderJson(feedHtml);
```

Comment:
given the format we're dealing with here, and to prevent any confusion, can we rename the parameter to feedJSON?

```suggestion
    renderFeed(function (feedJSON) {
      lib.renderJson(feedJSON);
```